lib/action_controller/serialization.rb

```diff
@@ -25,6 +25,11 @@ def get_serializer(resource, options = {})
           "Please pass 'adapter: false' or see ActiveSupport::SerializableResource#serialize"
         options[:adapter] = false
       end
+      if resource.respond_to?(:current_page) && resource.respond_to?(:total_pages)
+        options[:pagination] = {}
+        options[:pagination][:original_url] = original_url
+        options[:pagination][:query_parameters] = query_parameters
+      end
```

Comment:
Hey @bf4 , Is the idea to send the whole request through objects?
We are using just `original_url` in case of it is not already defined in `options[:links][:self]` and `query_parameters `, I really don't think it is good idea, but may have an issue that I am seeing that you are. 

It might be an option if [this feature](https://github.com/rails-api/active_model_serializers/issues/1062) will be approved.